realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -52,14 +53,20 @@
     private final NativeContext context;
 
     private final OsSharedRealm sharedRealm;
+    private final @Nullable OsKeyPathMapping osKeyPathMapping;
 
-    Table(OsSharedRealm sharedRealm, long nativeTableRefPointer) {
+    Table(OsSharedRealm sharedRealm, long nativeTableRefPointer, @Nullable OsKeyPathMapping osKeyPathMapping) {
         this.context = sharedRealm.context;
         this.sharedRealm = sharedRealm;
         this.nativeTableRefPtr = nativeTableRefPointer;
+        this.osKeyPathMapping = osKeyPathMapping;
```

Comment:
Not sure it is safe to store it like this? The keypath mapping can be updated by a callback from Core, making storing it here dangerous. Unless the usage is very shortlived. But maybe I am missing something?

Reply:
Great catch, it is not safe as it can be recreated by a callback from core.